substrate/frame/bags-list/src/lib.rs

```diff
@@ -423,19 +497,64 @@ impl<T: Config<I>, I: 'static> SortedListProvider<T::AccountId> for Pallet<T, I>
 	}
 
 	fn on_insert(id: T::AccountId, score: T::Score) -> Result<(), ListError> {
-		List::<T, I>::insert(id, score)
+		match LockOrder::<T, I>::get() {
+			// insert node at the tail of the list to not affect the order of the list while
+			// ordering lock is set.
+			Some(_) => List::<T, I>::insert_force_lowest(id, score),
+			None => List::<T, I>::insert(id, score),
+		}
 	}
 
 	fn get_score(id: &T::AccountId) -> Result<T::Score, ListError> {
 		List::<T, I>::get_score(id)
 	}
 
 	fn on_update(id: &T::AccountId, new_score: T::Score) -> Result<(), ListError> {
-		Pallet::<T, I>::do_rebag(id, new_score).map(|_| ())
+		match LockOrder::<T, I>::get() {
+			Some(_) => {
+				// lock is set, on_update is a noop.
+				frame_support::ensure!(
+					list::Node::<T, I>::get(&id).is_some(),
+					ListError::NodeNotFound
+				);
+				Ok(())
+			},
+			None => Pallet::<T, I>::do_rebag(id, new_score).map(|_| ()),
+		}
 	}
 
 	fn on_remove(id: &T::AccountId) -> Result<(), ListError> {
-		List::<T, I>::remove(id)
+		match LockOrder::<T, I>::get() {
+			Some(_) => {
+				// lock is set, on_remove is a noop.
+				frame_support::ensure!(
+					list::Node::<T, I>::get(&id).is_some(),
+					ListError::NodeNotFound
+				);
+				Ok(())
+			},
+			None => List::<T, I>::remove(id),
+		}
+	}
+
+	fn lock_ordering() -> Result<(), ListError> {
```

Comment:
Should these be in a separate trait? 
Also an observation, this trait is called `SortedListProvider` but when locked it is not really a sorted list provider.